Orange/regression/pls.py

```diff
@@ -0,0 +1,190 @@
+import numpy as np
+import sklearn.cross_decomposition as skl_pls
+
+from Orange.data import Table, Domain, Variable, \
+    ContinuousVariable, StringVariable
+from Orange.data.util import get_unique_names, SharedComputeValue
+from Orange.preprocess.score import LearnerScorer
+from Orange.regression.base_regression import SklLearnerRegression, \
+    SklModelRegression
+
+__all__ = ["PLSRegressionLearner"]
+
+
+class _FeatureScorerMixin(LearnerScorer):
+    feature_type = Variable
+    class_type = ContinuousVariable
+
+    def score(self, data):
+        model = self(data)
+        return np.abs(model.coefficients), model.domain.attributes
+
+
+class _PLSCommonTransform:
+
+    def __init__(self, pls_model):
+        self.pls_model = pls_model
+
+    def _transform_with_numpy_output(self, X, Y):
+        """
+        # the next command does the following
+        x_center = X - pls._x_mean
+        y_center = Y - pls._y_mean
+        t = x_center @ pls.x_rotations_
+        u = y_center @ pls.y_rotations_
+        """
+        pls = self.pls_model.skl_model
+        t, u = pls.transform(X, Y)
+        return np.hstack((t, u))
+
+    def __call__(self, data):
+        if data.domain != self.pls_model.domain:
+            data = data.transform(self.pls_model.domain)
+        if len(data.Y.shape) == 1:
+            Y = data.Y.reshape(-1, 1)
+        else:
+            Y = data.Y
+        return self._transform_with_numpy_output(data.X, Y)
+
+    def __eq__(self, other):
```

Comment:
Here you are offloading comparisons to `pls_model`, which makes sense, so it is cool.

But these `__eq__`  and `__hash__` need to be tested in a way that actually tests the comparison of two build domains (so, in turn, also `PLSProjector`). See `test_eq_hash` in `test_pca.py`.